### PR TITLE
[SPARK-12633][Python][MLlib][DOC] Update param descriptions in regression.py

### PR DESCRIPTION
**Pull request by vijaykiran**
Updates the param, in python mllib's regression.py, descriptions to be consistent. See [SPARK-11219] for
more details.

### CODE REVIEW

**Comment by jkbradley** [comment]
I just added a note to the parent JIRA about a formatting issue affecting all 5 PRs: [https://issues.apache.org/jira/browse/SPARK-11219?focusedCommentId=15090225&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15090225]
Could you please check it out & ping when I should review again?  Thank you!

**Comment by vijaykiran** [comment]
ping @jkbradley !

File: python/pyspark/mllib/regression.py [hunk 1]
```diff
@@ -37,10 +37,11 @@ class LabeledPoint(object):
...
 
-    :param label: Label for this data point.
-    :param features: Vector of features for this point (NumPy array,
-            list, pyspark.mllib.linalg.SparseVector, or scipy.sparse
-            column matrix)
+    :param label:
+      Label for this data point.
+    :param features:
+      Vector of features for this point (NumPy array, list,
+      pyspark.mllib.linalg.SparseVector, or scipy.sparse column matrix)
```
Comment (BryanCutler) [inline python/pyspark/mllib/regression.py]: Capitalize the first word and add a period at the end

File: python/pyspark/mllib/regression.py [hunk 2]
```diff
@@ -66,8 +67,10 @@ class LinearModel(object):
     """
     A linear model that has a vector of coefficients and an intercept.
 
-    :param weights: Weights computed for every feature.
-    :param intercept: Intercept computed for this model.
+    :param weights:
+      Weights computed for every feature.
```
Comment (BryanCutler) [inline python/pyspark/mllib/regression.py]: Capitalize the first word

File: python/pyspark/mllib/regression.py [hunk 3]
```diff
@@ -393,27 +401,35 @@ def train(cls, data, iterations=100, step=1.0, regParam=0.01,
...
+          Fraction of data to be used for each SGD iteration.
+          (default: 1.0)
+        :param initialWeights:
+          The initial weights.
+          (default: None)
+        :param intercept:
+          Boolean parameter which indicates the use or not of the
+          augmented representation for training data (i.e. whether bias
+          features are activated or not).
+         (default: False)
```
Comment (BryanCutler) [inline python/pyspark/mllib/regression.py]: not aligned properly

**Comment by BryanCutler** [comment]
Thanks @vijaykiran , just a couple minor corrections and would you mind changing the PR title to indicate this is for regression.py?

File: python/pyspark/mllib/regression.py (hunk 1 above)
Comment (vijaykiran) [inline python/pyspark/mllib/regression.py]: Do you mean "pyspark.mllib" should be "Pyspark" ? I thought package names should be left alone.

File: python/pyspark/mllib/regression.py (hunk 2 above)
Comment (vijaykiran) [inline python/pyspark/mllib/regression.py]: Sorry - I don't understand, which word ? "Weights" is already capitalized ...

File: python/pyspark/mllib/regression.py (hunk 1 above)
Comment (BryanCutler) [inline python/pyspark/mllib/regression.py]: Sorry, my bad - it is already capitalized, my font was too small I guess :).  Could you just add a period after the end of the description?  So should read ..."scipy.sparse column matrix)."

File: python/pyspark/mllib/regression.py (hunk 2 above)
Comment (BryanCutler) [inline python/pyspark/mllib/regression.py]: My fault again, this is fine

File: python/pyspark/mllib/regression.py (hunk 3 above)
Comment (vijaykiran) [inline python/pyspark/mllib/regression.py]: Fixed!

**Comment by vijaykiran** [comment]
@BryanCutler Fixed the indentation and added period.

**Comment by mengxr** [comment]
ok to test

**Comment by BryanCutler** [comment]
we need to settle on the `:Allowed values:` format, also in classification.py, before we can finish this one up.

**Comment by MLnick** [comment]
@BryanCutler it looks like a version of the "allowed values" got merged in the PR for `classification.py` - https://github.com/apache/spark/pull/11183/files#diff-3a683f424bec70006f6c225a32d162f9R297?

**Comment by BryanCutler** [comment]
Yeah, we were discussing the format of "allowed values" in that PR.  I just wanted to make sure we used the same format for the similar params here.  I'm not too crazy about the extra blank lines, but Sphinx requires them to format it correctly without generating errors.

**Comment by MLnick** [comment]
@BryanCutler ok, from that comment discussion it seems the format agreed would be:

```
Allowed values:

- "l1" ...
- "l2" ...
```

As we discussed in the PR for `tree.py`, I prefer `Supported values` over `Allowed values`, though that would require updating it here and in `classification.py` (and #10598 has been merged already).

**Comment by BryanCutler** [comment]
@MLnick I don't have a preference for `Supported values` vs `Allowed values` but I would like to see it consistent throughout, so it should probably be updated in `classification.py`.  Should I do that here, or open another issue for that?

**Comment by MLnick** [comment]
It's pretty minor so just add it here

**Pull request closed without merging**